#!/usr/bin/env python3
PACKAGE = "pid_carrot_follower"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("L_Kp", double_t, 0, "L_Kp", 1.0, 0.1, 5.0)
gen.add("L_Ki", double_t, 0, "L_Kp", 0.0, 0.0, 5.0)
gen.add("L_Kd", double_t, 0, "L_Kp", 0.0, 0.0, 5.0)

gen.add("M_Kp", double_t, 0, "L_Kp", 1.0, 0.1, 5.0)
gen.add("M_Ki", double_t, 0, "L_Kp", 0.0, 0.0, 5.0)
gen.add("M_Kd", double_t, 0, "L_Kp", 0.0, 0.0, 5.0)

gen.add("H_Kp", double_t, 0, "H_Kp", 1.0, 0.1, 5.0)
gen.add("H_Ki", double_t, 0, "H_Ki", 0.0, 0.0, 5.0)
gen.add("H_Kd", double_t, 0, "H_Kd", 0.0, 0.0, 5.0)

gen.add("M_speed", double_t, 0, "M_speed", 2.0, 0.0, 10.0)
gen.add("H_speed", double_t, 0, "H_speed", 5.0, 0.0, 10.0)

gen.add("speed_change_treshold", double_t, 0, "speed_change_treshold", 0.1, 0.0, 0.5)

exit(gen.generate(PACKAGE, "pid_carrot_follower", "PIDTuner"))
